src/annotations/AnnotationManager.php

```diff
@@ -299,6 +306,22 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
                 }
             }
 
+            if ($this->_traitsSupported && $member_name) {
```

Comment:
The existing if block checks `(isset($file) && isset($file->data[$key]))`, and `isset($file->data[$key])` will always be false if we need to look up the method in `$file`, so it doesn't make sense to move this code inside that block. But the patch doesn't access `$file` except inside a `if (isset($file->traitMethodOverrides[$class_name][$member_name]))` block, so if `$file` is not set, no warning will be generated.